packages/zosjobs/__tests__/__system__/GetJobs.system.test.ts

```diff
@@ -179,894 +179,991 @@ describe("Get Jobs - System Tests", () => {
+                            expect(content).toContain("NUMBER OF RECORDS PROCESSED WAS 3");
+                            break;
+                        }
+                    }
+                    testEnvironment.resources.jobs.push(job);
+                }, LONG_TIMEOUT);
+
+                it("Should get spool content for a single job with encoding", async () => {
+                    const idcams = fs.readFileSync(join(TEST_RESOURCES_DIR, "jcl/instream_rexx_content.jcl")).toString();
+                    const DATA_TO_CHECK = "PUTTYPUTTYPUTTYPUTTY¬¬";
+                    const renderedJcl = TextUtils.renderWithMustache(idcams,
+                        {JOBNAME: MONITOR_JOB_NAME, ACCOUNT, JOBCLASS, TYPERUNPARM: "", SYSAFF, CONTENT: DATA_TO_CHECK});
+                    const NUM_OF_SPOOL_FILES = 4;
+                    const DD_WITH_CONTENT = "SYSTSPRT";
+                    const job = await SubmitJobs.submitJclNotify(REAL_SESSION, renderedJcl);
+                    const files = await GetJobs.getSpoolFilesForJob(REAL_SESSION, job);
+                    expect(files.length).toBe(NUM_OF_SPOOL_FILES); // verify expected number of DDs
+                    for (const file of files) {
+                        if (file.ddname === DD_WITH_CONTENT) {
+                            const content = await GetJobs.getSpoolContentById(REAL_SESSION, job.jobname, job.jobid, file.id, "IBM-037");
+                            expect(content).toContain("NUMBER OF RECORDS PROCESSED WAS 3");
+                            expect(content).toContain("PUTTYPUTTYPUTTY");
+                            expect(content).not.toContain("¬¬");
+                            expect(content).toContain("^^");
+                            break;
+                        }
+                    }
+                    testEnvironment.resources.jobs.push(job);
+                }, LONG_TIMEOUT);
+            });
         });
-    });
 
-    describe("download JCL", () => {
-        it("should be able to get jcl from a job that was submitted", async () => {
-            const job = await SubmitJobs.submitJcl(REAL_SESSION, JCL);
-            const jcl = await GetJobs.getJclForJob(REAL_SESSION, job);
-            expect(jcl).toContain("EXEC PGM=IEFBR14");
-            expect(jcl).toContain("JOB");
+        /**********************************************/
+        // API methods "getJcl..." system tests
+        describe("Get JCL APIs", () => {
+            describe("invalid request error handling", () => {
+                it("should detect and surface an error for getting JCL that doesnt exist", async () => {
+                    const job = await SubmitJobs.submitJcl(REAL_SESSION, JCL);
+
+                    await DeleteJobs.deleteJobForJob(REAL_SESSION, job);
+
+                    // Add a delay to ensure job deletion is recognized
+                    await new Promise(resolve => setTimeout(resolve, 3000)); // Wait for 3 seconds
```

Comment:
Referring back to the `sleep` comment...
we could definitely make use of it here as well 😋